PatchNotes/Altibase_7.1/eng/Altibase_7_1_0_9_3_Patch_Notes.md

```diff
@@ -0,0 +1,585 @@
+
+-   **How to reproduce this bug**
+
+    -   **Reproduction conditions**
+
+    -   **Actual Results**
+
+    -   **Expected Results**
+
+-   **Workaround**
+
+-   **Changes**
+
+    -   Performance view
+    -   Property
+    -   Compile Option
+    -   Error Code
+
+### BUG-50758<a name=bug-50758></a> Fix an escape processing omission for double quote characters in APRE.
+
+-   **module** : mm-apre
+
+-   **Category** : Functional Error
+
+-   **Reproducibility** : Always
+
+-   **Description** : Escape character is automatically added to .c and .cpp files generated after apre precompile when EXEC SQL statements include "(double quote) character.
+    
+    Example)
+
+    .sc File : EXEC SQL SELECT ' " ' FROM DUAL;
+
+    ==\>
+
+    .c File : ulpSqlstmt.stmt = (char \*)"SELECT ' \\" ' FROM DUAL";
+
+- **How to reproduce this bug**
+
+  - **Reproduction conditions**
+
+    ```c
+    int main()
+    {
+        EXEC SQL SELECT ' " ' FROM DUAL;
+    }
+    ```
+
+  - **Actual Results**
+
+    It is converted as follows in the .c file.
+
+    ```c
+    int main()
+    {
+        /*  SELECT ' " ' FROM DUAL; */
+    {
+        struct ulpSqlstmt ulpSqlstmt;
+        memset(&ulpSqlstmt, 0, sizeof(ulpSqlstmt));
+        ulpSqlstmt.stmttype = 4;
+        ulpSqlstmt.stmtname = NULL;
+        ulpSqlstmt.ismt = 0;
+        ulpSqlstmt.numofhostvar = 0;
+        ulpSqlstmt.statusptr = NULL;
+        ulpSqlstmt.errcodeptr = NULL;
+        ulpSqlstmt.isatomic = 0;
+        ulpSqlstmt.stmt = (char *)"SELECT ' " ' FROM DUAL";
+     ...
+    }
+    }
+    ```
+
+  -   **Expected Results**
+
+      In the .c file, users can find \ is added before " as follows.
+      
+      ```c
+      int main()
+      {
+          /*  SELECT ' " ' FROM DUAL; */
+      {
+          struct ulpSqlstmt ulpSqlstmt;
+          memset(&ulpSqlstmt, 0, sizeof(ulpSqlstmt));
+          ulpSqlstmt.stmttype = 4;
+          ulpSqlstmt.stmtname = NULL;
+          ulpSqlstmt.ismt = 0;
+          ulpSqlstmt.numofhostvar = 0;
+          ulpSqlstmt.statusptr = NULL;
+          ulpSqlstmt.errcodeptr = NULL;
+          ulpSqlstmt.isatomic = 0;
+          ulpSqlstmt.stmt = (char *)"SELECT ' \" ' FROM DUAL";
+      ...
+      }
+      }
+      ```
+
+-   **Workaround**
+
+-   **Changes**
+
+    -   Performance view
+    -   Property
+    -   Compile Option
+    -   Error Code
+
+### BUG-50778<a name=bug-50778></a> Add exception handling to the getColumnCount function.
+
+-   **module** : qp
+-   **Category** : Fatal
+-   **Reproducibility** : Rare
+-   **Description** : Altibase internally invokes the getColumnCount function to request column information during the Prepare process. There was an issue that the server was abnormally terminated because of incorrect data structure access in an exceptional situation. Altibase added exception handling to this situation.
```

Comment:
```suggestion
-   **Description** : Altibase internally invokes the getColumnCount function to request column information during the Prepare process. There was an issue that the server was abnormally terminated because of an invalid data structure access in an exceptional situation. Altibase added exception handling to this situation.
```